spec/arc-spec.js

```diff
@@ -193,6 +193,99 @@ describe('c3 chart arc', function () {
+                        columns: [
+                            ['padded1', 100],
+                            ['padded2', 90],
+                            ['padded3', 50],
+                            ['padded4', 20]
+                        ]
+                    },
+                    type: 'gauge',
+                    color: {
+                        pattern: ['#FF0000', '#F97600', '#F6C600', '#60B044'],
+                        threshold: {
+                            values: [30, 80, 95]
+                        }
+                    }
+                };
+            });
+            var arcColor = ['#60b044', '#f6c600', '#f97600', '#ff0000'];
+
+            describe('should contain arcs ', function () {
+                it('each data_column should have one arc', function () {
+                    chart.internal.main.selectAll('.c3-chart-arc .c3-arc').each(function (d, i) {
+                        expect(d3.select(this).classed('c3-chart-data-' + args.data.columns[i][0])).toBeTruthy();
+                    });
+                });
+
+                it('each arc should have the color from color_pattern if color_treshold is given ', function () {
+                    chart.internal.main.selectAll('.c3-chart-arc .c3-arc').each(function (d, i) {
+                        expect(d3.select(this).style('fill')).toBe(arcColor[i]);
+                    });
+                });
+            });
+
+            describe('should contain backgrounds ', function () {
+                it('each data_column should have one background', function () {
+                    chart.internal.main.selectAll('.c3-chart-arcs path.c3-chart-arcs-background').each(function (d, i) {
+                        expect(d3.select(this).classed('c3-chart-arcs-background-'+ i)).toBeTruthy();
+                    });
+                });
+
+                it('each background should have tbe same color', function () {
+                    chart.internal.main.selectAll('.c3-chart-arcs path.c3-chart-arcs-background').each(function () {
+                        expect(d3.select(this).style('fill')).toBe('#e0e0e0');
+                    });
+                });
+            });
+
+            describe('should contain labels', function () {
+                it('each data_column should have a label', function () {
+                    chart.internal.main.selectAll('.c3-chart-arc .c3-gauge-value').each(function (d, i) {
+                        expect(d3.select(this).text()).toBe(args.data.columns[i][1]);
+                    });
+                });
+
+                it('each label should have the same color', function () {
+                    chart.internal.main.selectAll('.c3-chart-arc .c3-gauge-value').each(function () {
+                        expect(d3.select(this).style('fill')).toBe('#000');
+                    });
+
+                });
+
+                it('if only one data_column is visible the label should have "" for transform', function () {
```

Comment:
@GoekhanOezen
`done` is missing here and that causes uncaught error in a following test case which is running 1 second later of this test case.